src/js/modules/supervisors/Repository.ts

```diff
@@ -208,7 +210,10 @@ class Repository {
                   recordBatch,
                   resultMap
                 )
-              );
+              )
+              .catch((responseError: ProcessBatchReplyItem) => {
+                return responseError;
```

Comment:
Isn't this the same as not having this `catch` callback at all?

Reply:
No, because if I don't handle that expectation, that expects a `ProcessBatchReplayItem` it could be up to another catch that doesn't expect this type, like the next one `ln:218`

Reply:
Ahh. You're returning it, not throwing it. Goooot it.